File: java/src/test/java/codingdojo/DailyForecastsTest.java
package codingdojo;

import org.approvaltests.Approvals;
import org.approvaltests.core.Options;
import org.approvaltests.scrubbers.DateScrubber;
import org.approvaltests.scrubbers.RegExScrubber;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

public class DailyForecastsTest {

    final ByteArrayOutputStream capturedOutput = new ByteArrayOutputStream();
    PrintStream originalOutput;

    @BeforeEach
    void captureStandardOutput() {
        originalOutput = System.out;
        System.setOut(new PrintStream(capturedOutput));
    }

    @AfterEach
    void resetStandardOutput() {
        System.setOut(originalOutput);
    }

    @Test
    void salesForecast() {
        var forecasts = new DailyForecasts();
        forecasts.printSalesForecasts();
        Approvals.verify(capturedOutput);
    }
}

File: kotlin/src/test/kotlin/org/codingdojo/DailyForecastsTest.kt
package org.codingdojo

import org.approvaltests.Approvals
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.io.ByteArrayOutputStream
import java.io.PrintStream

class DailyForecastsTest {

    val capturedOutput = ByteArrayOutputStream()
    var originalOutput: PrintStream? = null

    @BeforeEach
    fun captureStandardOutput() {
        originalOutput = System.out
        System.setOut(PrintStream(capturedOutput))
    }

    @AfterEach
    fun resetStandardOutput() {
        System.setOut(originalOutput)
    }

    @Test
    fun salesForecast() {
        val forecasts = DailyForecasts()
        forecasts.printSalesForecasts()
        Approvals.verify(capturedOutput)
    }
}
